### PR TITLE
Fix rbac after controller-runtime update

### PR DESCRIPTION
**Pull request by alvaroaleman**
In #19440 controller-runtime was updated, changing its default for the
leader election to be both a configmap (which were never intended to be
used for this) and the lease resource. The latter requires additional
rbac which is added here.

The reason for using two apigroups for this in parallel is to allow safe
transitions. We should after some grace period start using the lease
resource only.

### CODE REVIEW

**Comment by chaodaiG** [comment]
@michelle192837 for awareness

**Review by michelle192837 — APPROVED** [review]
/hold

LGTM from my perspective, but holding in case you want review from one of the assigned folks as well. Hold cancel at will if not.

**Comment by alvaroaleman** [comment]
> LGTM from my perspective, but holding in case you want review from one of the assigned folks as well. Hold cancel at will if not

No its fine, its a copy of what we already deployed in Openshift

/hold cancel

**Comment by alvaroaleman** [comment]
@chaodaiG I had to rebase because there was a mergeconflict in ANNOUNCEMENTS.md, could you give a new lgtm?

**Pull request merged**
